In [1]:
import os
os.getcwd()

'C:\\Users\\Lamia'

# CH.1 language processing in python

“from NLTK’s book module, load all items.” The book module contains all the data
you will need as you read this chapter. After printing a welcome message, it loads the
text of several books (this will take a few seconds).

In [2]:
from nltk.book import *

*** Introductory Examples for the NLTK Book ***
Loading text1, ..., text9 and sent1, ..., sent9
Type the name of the text or sentence to view it.
Type: 'texts()' or 'sents()' to list the materials.
text1: Moby Dick by Herman Melville 1851
text2: Sense and Sensibility by Jane Austen 1811
text3: The Book of Genesis
text4: Inaugural Address Corpus
text5: Chat Corpus
text6: Monty Python and the Holy Grail
text7: Wall Street Journal
text8: Personals Corpus
text9: The Man Who Was Thursday by G . K . Chesterton 1908


let’s try generating some random text

In [3]:
text9.generate()

Building ngram index...


is , and finding a Sunday teashop , turned in his loose straw hat on
the bench under the advice of his hands . , or an apple tree , but not
an anarchist . already irritated nerves ; and Buddhism is not a man
worth listening to , even if one only laughed at by something at least
he was -- blood running down his knife and fork . " " I ' d better ."
the foundation of solid meekness which lies at the anarchist breakfast
and had been too true . Sunday always insists on the foremost


'is , and finding a Sunday teashop , turned in his loose straw hat on\nthe bench under the advice of his hands . , or an apple tree , but not\nan anarchist . already irritated nerves ; and Buddhism is not a man\nworth listening to , even if one only laughed at by something at least\nhe was -- blood running down his knife and fork . " " I \' d better ."\nthe foundation of solid meekness which lies at the anarchist breakfast\nand had been too true . Sunday always insists on the foremost'

## What is a token?
token is the technical name for a sequence of characters—such as hairy, his, or :)—that we want
to treat as a group

In [4]:
len(text9)# number of tokens

69213

So Genesis has 69,213 words and punctuation symbols, or “tokens.”

In [5]:
Tokens=set(text9) #list of all words, punctuations and symbols
Tokens#list of all words, punctuations and symbols

{'once',
 'torments',
 'abandoned',
 'funk',
 'If',
 'ask',
 'thrill',
 'May',
 'plunge',
 'mad',
 'against',
 'infinity',
 'emptied',
 'voiceless',
 'architecture',
 'music',
 'than',
 'formulate',
 'Insulted',
 'pure',
 'Consequently',
 'shops',
 'Tupper',
 'Dynamiters',
 'obstacle',
 'marked',
 'pair',
 'clay',
 'wandering',
 'sheath',
 'flask',
 'opinions',
 'parcel',
 'forms',
 'astonishing',
 'when',
 'position',
 'appear',
 'beetle',
 'don',
 'postman',
 'feared',
 'performed',
 'dresses',
 'portion',
 'TWO',
 'tyrants',
 'luck',
 'killing',
 'tonight',
 'hurt',
 'energetic',
 'prevailed',
 'Supreme',
 'miserable',
 'pursue',
 'Could',
 'gravity',
 'ourang',
 'change',
 'spike',
 'Breakfast',
 'district',
 'vanished',
 'hence',
 ".'",
 'lighted',
 'troubadour',
 'launch',
 'later',
 'affably',
 'Satanic',
 'vizards',
 'triangular',
 'right',
 'courtesy',
 'Talk',
 'control',
 'feverishly',
 'palsy',
 'hypocrite',
 'VIII',
 'thorns',
 'timid',
 'remembers',
 'lets',
 'scepticism'

In [6]:
Sorted_tokens=sorted(Tokens)
Sorted_tokens

['!',
 '!"',
 "!'",
 '!--',
 '"',
 '"\'',
 '"--',
 '";',
 "'",
 "'--",
 "'.",
 '\'."',
 "';",
 "'?",
 '\'?"',
 '(',
 ')',
 '),',
 ').',
 ')."',
 '*',
 ',',
 ',"',
 ",'",
 ',\'"',
 '-',
 '--',
 '--"',
 "--'",
 '.',
 '."',
 '."\'',
 ".'",
 '.\'"',
 '.--',
 '1350',
 '1908',
 '217',
 '45',
 '7',
 ':',
 ':--',
 ';',
 '?',
 '?"',
 "?'",
 '?)',
 'A',
 'ACCUSER',
 'ACROSS',
 'ANARCHY',
 'AS',
 'AT',
 'About',
 'Above',
 'Absurd',
 'Accepting',
 'Acheron',
 'Across',
 'Acting',
 'Adam',
 'Admiral',
 'Aesop',
 'African',
 'After',
 'Afterwards',
 'Again',
 'Against',
 'Ages',
 'Ah',
 'Albany',
 'Albert',
 'Alhambra',
 'Alice',
 'All',
 'Ally',
 'Almighty',
 'Almost',
 'Along',
 'Alpine',
 'Also',
 'Am',
 'America',
 'American',
 'An',
 'Anarch',
 'Anarchist',
 'Anarchists',
 'Anarchy',
 'And',
 'Anne',
 'Another',
 'Answer',
 'Anyhow',
 'Archdeacon',
 'Are',
 'Argus',
 'Aristocrats',
 'Armageddon',
 'As',
 'Asiatic',
 'Asked',
 'Assyrian',
 'At',
 'Attila',
 'B',
 'BEFORE',
 'BURGUNDY',
 'Baal',

In [7]:
len(Tokens) #size of unique types (words,symbols and puncs.)

6807

Calculating lexical richness factor معامل الثراء المعجمي : 
Average # of used words


In [8]:
len(text9) / len(Tokens)

10.167915381225209

In [9]:
text9.count("the") #howmany times the word the appeared 

3291

In [10]:
(text9.count("the")/len(text9))*100 #But be careful, here you considered symbols and punctuations as words

4.754887087685839

In [11]:
last_four_tokens=Sorted_tokens[-4:]
last_four_tokens

['zay', 'ze', 'zem', 'zmite']

In [12]:
join_last_four_tokens=' '.join(last_four_tokens)
join_last_four_tokens

'zay ze zem zmite'

In [13]:
split_last_four_tokens=join_last_four_tokens.split()
split_last_four_tokens

['zay', 'ze', 'zem', 'zmite']

# Frequency destribution

In [14]:
fdist9 = nltk.FreqDist(text9)
fdist9 #dictionary

NameError: name 'nltk' is not defined

In [ ]:
fdist9['the']

In [ ]:
fdist9.keys()

In [ ]:
fdist9.values()

In [ ]:
fdist9.plot(50, cumulative=True,title='first 50 frequencies')

In [ ]:
fdist9.tabulate()

Selecting words with one to three letters

In [ ]:
Doc = set(text2)
some_words = [w for w in Doc if len(w) in range(1,4)]
sorted(some_words)


## Collocations and Bigrams

A collocation is a sequence of words that occur together unusually often. Thus redwine is a collocation, whereas the wine is not

In [15]:
text4.collocations()

United States; fellow citizens; four years; years ago; Federal
Government; General Government; American people; Vice President; God
bless; Chief Justice; Old World; Almighty God; Fellow citizens; Chief
Magistrate; every citizen; one another; fellow Americans; Indian
tribes; public debt; foreign nations


In [16]:
#What do you think the output of this code?
max([len(w) for w in text1])

20

In [17]:
[w for w in text1 if len(w)==20]

['uninterpenetratingly']

[w for w in text if condition]
Conditions 
<img src="Capture.png">

In [18]:
#[w for w in text9 if w.startswith('L')]

In [19]:
#[w for w in text9 if w.isdigit()]

In [20]:
ALNUM_LIST=[w for w in text9 if w.isalnum()]

In [21]:
ALNUM_LIST[:3]

['The', 'Man', 'Who']

In [22]:
#[w.upper() for w in text9]